openeo/extra/job_management.py

```diff
@@ -337,43 +358,39 @@ def run_jobs(
         if job_db.exists():
             # Resume from existing db
             _log.info(f"Resuming `run_jobs` from existing {job_db}")
-            df = job_db.read()
-            status_histogram = df.groupby("status").size().to_dict()
-            _log.info(f"Status histogram: {status_histogram}")
-
-        df = self._normalize_df(df)
+            #df = job_db.read()
+            #status_histogram = df.groupby("status").size().to_dict()
+            #_log.info(f"Status histogram: {status_histogram}")
+        else:
+            df = self._normalize_df(df)
+            job_db.persist(df)
 
         while (
-            df[
-                # TODO: risk on infinite loop if a backend reports a (non-standard) terminal status that is not covered here
-                (df.status != "finished")
-                & (df.status != "skipped")
-                & (df.status != "start_failed")
-                & (df.status != "error")
-                & (df.status != "canceled")
-            ].size
-            > 0
+            job_db.has_unfinished_jobs()
+
         ):
 
             with ignore_connection_errors(context="get statuses"):
-                self._track_statuses(df)
-            status_histogram = df.groupby("status").size().to_dict()
-            _log.info(f"Status histogram: {status_histogram}")
-            job_db.persist(df)
+                self._track_statuses(job_db)
+            #TODO do we still want to support this? Would require a 'count_by_status'?
+            #status_histogram = df.groupby("status").size().to_dict()
+            #_log.info(f"Status histogram: {status_histogram}")
```

Comment:
maybe define a `get_status() -> str` method  on `JobDatabaseInterface` to move that responsibility out of run_jobs?

Reply:
I now opted to move it into the file based job db itself.
If a job db is backed by an actual database, it probably makes more sense to monitor progress in something more visual.